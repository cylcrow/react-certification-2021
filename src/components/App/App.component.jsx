import React from 'react';
import { LayoutWrapper } from '../Layout';

function App() {
  return <LayoutWrapper />;
}

export default App;
