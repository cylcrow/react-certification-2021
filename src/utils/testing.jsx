import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}