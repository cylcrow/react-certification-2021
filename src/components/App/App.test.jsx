import React from 'react';
import '../../utils/testing';
import { render, unmountComponentAtNode } from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { getByTestId } from '@testing-library/react';
import App from './index';

let container;
const build = () => {
  render(<App />, container);
  return {
    LayoutWrapper: () => getByTestId(container, 'layout-wrapper'),
  };
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('App layout', () => {
  it('renders', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('displays layout wrapper', () => {
    const { LayoutWrapper } = build();

    expect(LayoutWrapper()).toBeInTheDocument();
  });
});
