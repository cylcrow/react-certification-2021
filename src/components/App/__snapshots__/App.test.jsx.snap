// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App layout renders 1`] = `
<LayoutWrapper>
  <styled.div>
    <h1>
      Welcome to the challenge!
    </h1>
  </styled.div>
  <HomeVideos />
</LayoutWrapper>
`;
